Building C# Assemblies fails because of missing generated files

```
❯ ./modules/mono/build_scripts/build_assemblies.py --godot-output-dir=./bin --push-nupkgs-local ~/dev/LocalNugetSource
Running MSBuild:  /usr/bin/dotnet msbuild /home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp.sln /restore /t:Build /p:Configuration=Debug /p:NoWarn=1591 /p:ClearNuGetLocalCache=true /p:PushNuGetToLocalSource=/home/geo/dev/LocalNugetSource
MSBuild version 17.4.1+fedecea9d for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj(142,3): error MSB4019: The imported project "/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GeneratedIncludes.props" was not found. Confirm that the expression in the Import declaration "Generated/GeneratedIncludes.props" is correct, and that the file exists on disk.
/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/GodotSharpEditor.csproj(43,3): error MSB4019: The imported project "/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/Generated/GeneratedIncludes.props" was not found. Confirm that the expression in the Import declaration "Generated/GeneratedIncludes.props" is correct, and that the file exists on disk.
/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj(142,3): error MSB4019: The imported project "/home/geo/dev/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GeneratedIncludes.props" was not found. Confirm that the expression in the Import declaration "Generated/GeneratedIncludes.props" is correct, and that the file exists on disk.
  Godot.SourceGenerators.Internal -> /home/geo/dev/godot/modules/mono/glue/GodotSharp/Godot.SourceGenerators.Internal/bin/Debug/netstandard2.0/Godot.SourceGenerators.Internal.dll
```

I've been using the same script to build the dotnet version of godot for months:

```sh
#!/usr/bin/env bash

git tag -d last-build
git tag last-build

set -e

scons module_mono_enabled=yes
./bin/godot.linuxbsd.editor.x86_64.mono --headless --generate-mono-glue modules/mono/glue

#dotnet nuget add source ~/dev/LocalNugetSource --name LocalNugetSource
rm -rf ~/.local/share/godot/mono/GodotNuGetFallbackFolder
mkdir -p ~/.local/share/godot/mono/GodotNuGetFallbackFolder

./modules/mono/build_scripts/build_assemblies.py --godot-output-dir=./bin \
 --push-nupkgs-local ~/dev/LocalNugetSource

# Export templates
#scons target=template_debug module_mono_enabled=yes
#scons target=template_release module_mono_enabled=yes

```

It recently stared to fail so I tried to delete everything extra via `git clean -xfd` and now I'm unable to build the mono version.

_Originally posted by @geowarin in https://github.com/godotengine/godot/issues/75152#issuecomment-1513810811_


---

@geowarin Moving to a new issue, because it doesn't look like it has anything to do with that issue.

Please post your system configuration, in particular the output of `dotnet --info`. Additionally please post the full output of the `godot --generate-mono-glue` command.

--- a/editor/editor_help.cpp
+++ b/editor/editor_help.cpp
@@ -2240,7 +2240,7 @@ void EditorHelp::_gen_doc_thread(void *p_udata) {
 
 static bool doc_gen_use_threads = true;
 
-void EditorHelp::generate_doc() {
+void EditorHelp::generate_doc(bool p_use_cache) {
 	if (doc_gen_use_threads) {
 		// In case not the first attempt.
 		_wait_for_thread();
@@ -2256,7 +2256,7 @@ void EditorHelp::generate_doc() {
 		doc = memnew(DocTools);
 	}
 
-	if (first_attempt && FileAccess::exists(get_cache_full_path())) {
+	if (p_use_cache && first_attempt && FileAccess::exists(get_cache_full_path())) {
 		if (doc_gen_use_threads) {
 			thread.start(_load_doc_thread, nullptr);
 		} else {

--- a/editor/editor_help.h
+++ b/editor/editor_help.h
@@ -204,7 +204,7 @@ protected:
 	static void _bind_methods();
 
 public:
-	static void generate_doc();
+	static void generate_doc(bool p_use_cache = true);
 	static DocTools *get_doc_data();
 	static void cleanup_doc();
 	static String get_cache_full_path();

--- a/modules/mono/editor/bindings_generator.cpp
+++ b/modules/mono/editor/bindings_generator.cpp
@@ -3894,7 +3894,7 @@ void BindingsGenerator::_log(const char *p_format, ...) {
 void BindingsGenerator::_initialize() {
 	initialized = false;
 
-	EditorHelp::generate_doc();
+	EditorHelp::generate_doc(false);
 
 	enum_types.clear();
 

